tests/unit/transfer_transaction_test.py

```
@@ -509,4 +527,389 @@ def test_approved_token_transfer_validation(mock_account_ids):
+            assert transfer.amount == -100
+        elif transfer.accountID.accountNum == account_id_2.num:
+            assert transfer.amount == 100
+
+
+def test_nft_transfer_with_approval_building(mock_account_ids):
+    """Test NFT transfers with approval flag are properly built."""
+    account_id_sender, account_id_recipient, node_account_id, token_id_1, _ = mock_account_ids
+    transfer_tx = TransferTransaction()
+
+    transfer_tx.add_nft_transfer(
+        NftId(token_id_1, 1), account_id_sender, account_id_recipient, False
```

Ruff (0.14.10)

Boolean positional value in function call

(FBT003)